src/Forms/GridField/GridField.php

```diff
@@ -362,11 +367,19 @@ public function FieldHolder($properties = array())
             'before' => true,
             'after' => true,
         );
+        $fragmentDeferred = [];
 
-        reset($content);
+        // TODO: Break the below into separate reducer methods
 
-        while (list($contentKey, $contentValue) = each($content)) {
-            if (preg_match_all('/\$DefineFragment\(([a-z0-9\-_]+)\)/i', $contentValue, $matches)) {
+        // Continue looping if any placeholders exist
```

Comment:
This block was a lot of no-fun to fix. =( Unlike other each() usages, it requires re-iteration of a single key multiple times.

Reply:
would the `array_filter()` pick up on the newly added items, would it be worth using `iterateVolatile` in the top `while` loop?

Reply:
Lets leave this for another time, there's a big top up top and tests are green